File: pset6/readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")

    l = count_letters(text)
    w = len(count_words(text))
    s = count_sentences(text)

    CLI = round((0.0588 * (l / (w * 0.01)) - 0.296 * (s / (w * 0.01)) - 15.8), 0)

    print(len(text))

    if (CLI < 1):
        print("Before Grade 1")
    elif (CLI > 16):
        print("Grade 16+")
    else:
        print(f"Grade {int(CLI)}")


def count_letters(text):
    letters = 0
    for j in text:
        if j.isalpha():
            letters += 1

    return letters


def count_words(text):
    words = text.split()
    return words


def count_sentences(text):
    sentences = 0
    for k in text:
        if k == "." or k == "!" or k == "?":
            sentences += 1

    return sentences


if __name__ == "__main__":
    main()

File: pset2/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>


int count_letters(string t);
int count_words(string t);
int count_sentences(string t);

int main(void)
{
    string text = get_string("Insert a text: ");

    int letters1 = count_letters(text);
    int words1 = count_words(text);
    int sentences1 = count_sentences(text);

    float index = (0.0588 * (letters1 / (words1 * 0.01))) - (0.296 * (sentences1 / (words1 * 0.01))) - 15.8;

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %.f\n", index);
    }
}


int count_letters(string t)
{
    int letters = 0;

    for (int i = 0, n = strlen(t); i < n; i++)
    {
        if ((t[i] >= 'A' && t[i] <= 'Z') || (t[i] >= 'a' && t[i] <= 'z'))
        {
            letters++;
        }
    }
    return letters;
}


int count_words(string t)
{
    int words = 0;

    for (int i = 0, n = strlen(t); i <= n; i++)
    {
        if (t[i] == 32 || t[i] == '\0')
        {
            words++;
        }
    }
    return words;
}


int count_sentences(string t)
{
    int sentences = 0;

    for (int i = 0, n = strlen(t); i < n; i++)
    {
        if (t[i] == '.' || t[i] == '!' || t[i] == '?')
        {
            sentences++;
        }
    }
    return sentences;
}